src/main/kotlin/com/exactpro/th2/lwdataprovider/db/CradleMessageExtractor.kt

```diff
@@ -122,7 +127,17 @@ class CradleMessageExtractor(configuration: Configuration, private val cradleMan
         val id = storedMessage.id.toString()
         val decodingStep = startStep("decoding")
         val tmp = createMessageDetails(id, 0, storedMessage) { decodingStep.finish() }
-        tmp.rawMessage = startStep("raw_message_parsing").use { RawMessage.parseFrom(storedMessage.content) }.also {
+        tmp.rawMessage = startStep("raw_message_parsing").use {
+            RawMessage.newBuilder().apply {
+                metadataBuilder.apply {
+                    putAllProperties(storedMessage.metadata?.toMap() ?: emptyMap())
+                    idBuilder.mergeFrom(storedMessage.id.toGrpcMessageId())
+                    timestampBuilder.mergeFrom(storedMessage.timestamp.toTimestamp())
+                    protocol = storedMessage.protocol ?: ""
+                }.build()
+                body = ByteString.copyFrom(storedMessage.content)
```

Comment:
We can guarantee that the `storedMessage.content` won't be changed so we can use `UnsafeByteOperations.unsafeWrap` instead of making a copy of that array